### PR TITLE
Add support for callbacks

### PR DESCRIPTION
**Pull request by ruimarinho**
This is a different take based on @dead-horse's pr.

### CODE REVIEW

File: withcallback.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
+
+/**
+ * Module dependencies.
+ */
+
+var Promise = require('native-or-bluebird')
+var assert = require('assert')
+var createCallback = require('./thenify').createCallback
+var promisify = require('./thenify').promisify
+var thenify = require('./thenify')
```
Comment (dead-horse) [inline withcallback.js]: why do we need this line?

Comment (ruimarinho) [inline withcallback.js]: Oops. It is not necessary.

**Comment by dead-horse** [comment]
finally landed this patch https://github.com/thenables/thenify/commit/2308efeb46764871a9733f87a765d8855328ef67

**Comment by dead-horse** [comment]
thanks!

**Comment by ruimarinho** [comment]
Awesome, thanks!

**Pull request closed without merging**